### ISSUE TITLE
add route not working, instead taking read

### ISSUE THREAD

**Issue by harikt** [auraphp/Aura.Router#33]
When we add a REST route

```
$router->attachResource('blog', '/blog');
```

http://localhost:8000/blog/1 

``` php
array(5) { ["controller"]=> string(4) "blog" ["action"]=> string(4) "read" ["id"]=> string(1) "1" ["format"]=> NULL ["REQUEST_METHOD"]=> string(3) "GET" } 
```

http://localhost:8000/blog/add pointing to same route with action `read` . 

``` php
array(5) { ["controller"]=> string(4) "blog" ["action"]=> string(4) "read" ["id"]=> string(3) "add" ["format"]=> NULL ["REQUEST_METHOD"]=> string(3) "GET" } 
```

I assume the `id` should be made as `\d`

**Comment by harikt** [comment]
Currently it is

``` php
// get a single resource by ID, optionally in a format
$router->addGet('read', '/{id}{format}')
    ->addTokens(array(
        'format' => '(\.[^/]+)?',
    ));

// get the form to add new resource
$router->addGet('add', '/add');
```

a hack to fix is 

``` php
// get a single resource by ID, optionally in a format
$router->addGet('read', '/{id}{format}')
    ->addTokens(array(
        'id' => '(\d)',
        'format' => '(\.[^/]+)?',
    ));

// get the form to add new resource
$router->addGet('add', '/add');
```

or move the `add` before read 

``` php
// get the form to add new resource
$router->addGet('add', '/add');

// get a single resource by ID, optionally in a format
$router->addGet('read', '/{id}{format}')
    ->addTokens(array(
        'id' => '(\d)',
        'format' => '(\.[^/]+)?',
    ));
```

I prefer having a `(\d)` than moving it . Thoughts ?

**Comment by harikt** [comment]
@pmjones ping .

**Comment by pmjones** [comment]
@harikt Sorry man. As usual, you are ahead and I am behind.

**Comment by harikt** [comment]
no problem Paul.

I know you are really busy.

_Hari K T_

You can ring me : +91 9388 75 8821

http://harikt.com , https://github.com/auraphp ,
http://www.linkedin.com/in/harikt , http://www.xing.com/profile/Hari_KT

Skype  : kthari85
Twitter : harikt

**Comment by harikt** [comment]
as this has been committed. I am closing it. Have not tested myself . Will open in case find the issue.

**Comment by pmjones** [comment]
Well, rats, I thought "fixes #33" in the commit log would close it.

**Comment by harikt** [comment]
aah, didn't I closed it ? Ha....
